### PR TITLE
Make _recordData lazy

### PR DESCRIPTION
**Pull request by hjdivad**
In particular, do not instantiate them for `findRecord`

### CODE REVIEW

File: addon/-private/system/snapshot.js [hunk 1]
```diff
@@ -65,7 +65,12 @@ export default class Snapshot {
      */
     this.modelName = internalModel.modelName;
 
-    this._changedAttributes = internalModel.changedAttributes();
+    if (internalModel.isLoading()) {
```
Comment (igorT) [inline addon/-private/system/snapshot.js]: It seems like this logic belongs in `internalModel`.

Comment (hjdivad) [reply addon/-private/system/snapshot.js]: @igorT seems fine; you'd prefer to push this into `internalModel.changedAttributes()`?

Comment (runspired) [inline addon/-private/system/snapshot.js]: I am skeptical. What about `reload` ?

Comment (hjdivad) [reply addon/-private/system/snapshot.js]: @runspired what's your preferred branch?  the state here is `root.loading`

**Comment by igorT** [comment]
Empty state should work

**Comment by hjdivad** [comment]
@igorT no, `root.loading` is not empty

**Comment by hjdivad** [comment]
we could change `root.loading` to `isEmpty: true` but i am far more wary of that than not eagerly calculating changed attributes during record reload

**Comment by hjdivad** [comment]
@igorT never mind, @runspired pointed me to the `isReloading` flag within internal model.

File: addon/-private/system/snapshot.js (hunk 1 above)
Comment (hjdivad) [reply addon/-private/system/snapshot.js]: I don't think it is though --- if you have an internal model for a `findRecord` (ie where `isLoading() && !isReloading`), what does it even mean conceptually to ask for changed attributes?

Comment (runspired) [reply addon/-private/system/snapshot.js]: If I reorder that you may see what I mean a little bit better: "logic specific to requests"

**Pull request merged**